h/models/group_scope.py

```diff
@@ -0,0 +1,34 @@
+# -*- coding: utf-8 -*-
+
+from __future__ import unicode_literals
+
+import sqlalchemy as sa
+
+from h._compat import urlparse
+from h.db import Base, mixins
+
+
+class GroupScope(Base):
+    __tablename__ = 'groupscope'
+    id = sa.Column(sa.Integer, autoincrement=True, primary_key=True)
+
+    #: A fully qualified domain name, e.g. example.com, www.nytimes.com or
+    #: web.hypothes.is.
+    hostname = sa.Column('hostname', sa.UnicodeText)
```

Comment:
I'm thinking that at first we'll support exact text match of the hostname part of the URL only (including subdomains). If we also need to look at the path part of the URL later, and maybe support pattern matching of some sort against the path part, we can add that later. (Maybe we'd add `path` as a separate column and start by retrieving the subset of `GroupScope`s whose `hostname` part is an exact string match, and then do a path part pattern match against the path parts of only that subset of `GroupScope`s.)

Reply:
So given the a URL from the client, to find the groups for that URL you'd first extract just the hostname part of the URL and then do:

```python
request.db.query(m.GroupScope).filter_by(hostname=u'www.nytimes.com').one().groups
```

(which may raise `NoResultFound`)